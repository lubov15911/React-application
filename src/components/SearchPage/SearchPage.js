import React, { Fragment } from 'react';

import HeaderContainer from "../HeaderContainer";
import SearchResultsContainer from '../SearchResultsContainer';

const SearchPage = () => (
    <Fragment>
        <HeaderContainer />
        <SearchResultsContainer />
    </Fragment>
);

export default SearchPage;
