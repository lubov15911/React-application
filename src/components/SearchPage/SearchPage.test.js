import React from 'react';
import { shallow } from 'enzyme';

import SearchPage from './SearchPage';

describe('SearchPage', () => {
    const items = [{
        id: 354912,
        poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
        title: 'Coco',
        release_date: '2017-10-27',
        genres: ["Adventure", "Comedy", "Family", "Animation"],
    }, {
        id: 333339,
        poster_path: 'https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg',
        title: 'Ready Player One',
        release_date: '2018-03-28',
        genres: ["Adventure", "Science Fiction", "Action"],
    }] ;
    const fakeFunction = () => {};

    let component;

    beforeEach(() => {
        component = shallow(<SearchPage
            films={items}
            routeToHomePage={fakeFunction}
            goToMoviePage={fakeFunction}
        />);
    });

    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should change searchOption state', () => {
        component.find('HeaderContainer').props().toggleSearch();
        expect(component.state().searchOption).toBe('Genre');
        component.find('HeaderContainer').props().toggleSearch();
        expect(component.state().searchOption).toBe('Title');
    });

    describe('should change searchResults to', () => {
        it('default array', () => {
            component.find('HeaderContainer').props().handleSearchValue('');
            expect(component.state().searchResults).toEqual(expect.arrayContaining(items));
        });

        it('search array', () => {
            component.find('HeaderContainer').props().handleSearchValue('ready');
            expect(component.state().searchResults).toEqual(expect.arrayContaining([items[1]]));
        });
    });
});
