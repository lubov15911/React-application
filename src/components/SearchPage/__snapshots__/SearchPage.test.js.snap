// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should render correctly 1`] = `
<Fragment>
  <HeaderContainer
    handleSearchValue={[Function]}
    movieData={null}
    searchOption="Title"
    toggleSearch={[Function]}
  />
  <SearchResultsContainer
    films={
      Array [
        Object {
          "genres": Array [
            "Adventure",
            "Comedy",
            "Family",
            "Animation",
          ],
          "id": 354912,
          "poster_path": "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
          "release_date": "2017-10-27",
          "title": "Coco",
        },
        Object {
          "genres": Array [
            "Adventure",
            "Science Fiction",
            "Action",
          ],
          "id": 333339,
          "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
          "release_date": "2018-03-28",
          "title": "Ready Player One",
        },
      ]
    }
    raiseClickEvent={[Function]}
    total={2}
  />
</Fragment>
`;
